File: oo-workshop/tsexercises/tests/unit/liskovsubstitution/transportationObject.test.ts
import {Car} from '../../../src/liskovsubstitution/car'
import {Bus} from '../../../src/liskovsubstitution/bus'
import {Bicycle} from '../../../src/liskovsubstitution/bicycle'
import {Skateboard} from '../../../src/liskovsubstitution/skateboard'


describe('transportation object', () => {
 
    it('when battery is charged, engine will start', () => {
        //given
        const car: Car = new Car()
        const bus: Bus = new Bus()

        //when
        const whencar: string = car.startEngine(true)
        const whenbus: string = bus.startEngine(true)

        //then
        expect(whencar).toContain("Starting engine")
        expect(whenbus).toContain("Starting engine")
    })

    it('when battery is not charged, engine will not start', () => {
        //given
        const car: Car = new Car()
        const bus: Bus = new Bus()

        //when
        const whencar: string = car.startEngine(false)
        const whenbus: string = bus.startEngine(false)

        //then
        expect(whencar).toContain("Charge battery")
        expect(whenbus).toContain("Charge battery")
    })

    it('when force applied, it will start moving', () => {
        //given
        const bicycle: Bicycle = new Bicycle()
        const skateboard: Skateboard = new Skateboard()
       
        //when
        const whenbicycle = bicycle.applyForce(true)
        const whenskateboard = skateboard.applyForce(true)

        //then
        expect(whenbicycle).toContain("Start moving")
        expect(whenskateboard).toContain("Start moving")
    })

    it('when force not applied, it will not start moving', () => {
        //given
        const bicycle: Bicycle = new Bicycle()
        const skateboard: Skateboard = new Skateboard()
        
        //when
        const whenbicycle = bicycle.applyForce(false)
        const whenskateboard = skateboard.applyForce(false)
      
        //then
        expect(whenbicycle).toContain("Apply more force")
        expect(whenskateboard).toContain("Apply more force")
    })

})

File: oo-workshop/mobprogrammingexercises/src/test/java/com/example/liskovsubstitution/mobprogramming/problem/TransportationObjectTest.java
package com.example.liskovsubstitution.mobprogramming.problem;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class TransportationObjectTest {
    @Test
    public void whenBatteryChargedIsTrueThenMessageContainsStartingEngine() {
        //given
        Car car = new Car();
        Bus bus = new Bus();

        //when
        String whencar = car.startEngine(true);
        String whenbus = bus.startEngine(true);

        //then
        assertTrue(whencar.contains("Starting engine"));
        assertTrue(whenbus.contains("Starting engine"));
    }

    @Test
    public void whenBatteryChargedIsFalseThenMessageContainsChargeBattery() {
        //given
        Car car = new Car();
        Bus bus = new Bus();

        //when
        String whencar = car.startEngine(false);
        String whenbus = bus.startEngine(false);

        //then
        assertTrue(whencar.contains("Charge battery"));
        assertTrue(whenbus.contains("Charge battery"));
    }

    @Test
    public void whenForceAppliedIsTrueThenMessageContainsStartMoving() {
        //given
        Bicycle bicycle = new Bicycle();
        Skateboard skateboard = new Skateboard();

        //when
        String whenbicycle = bicycle.applyForce(true);
        String whenskateboard = skateboard.applyForce(true);

        //then
        assertTrue(whenbicycle.contains("Start moving"));
        assertTrue(whenskateboard.contains("Start moving"));
    }

    @Test
    public void whenForceAppliedIsFalseThenMessageContainsApplyMoreForce() {
        //given
        Bicycle bicycle = new Bicycle();
        Skateboard skateboard = new Skateboard();

        //when
        String whenbicycle = bicycle.applyForce(false);
        String whenskateboard = skateboard.applyForce(false);

        //then
        assertTrue(whenbicycle.contains("Apply more force"));
        assertTrue(whenskateboard.contains("Apply more force"));
    }
}
